### PR TITLE
Flatten partitions tests into a single test

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:
- Move the "in-partition" acceptance tests and "x-partition" acceptance tests into a single test to enable faster test times.

How I've tested this PR:
- running it in CI 🤞 
- The above has been tested and the test runtime has decreased from ~42 mins to ~28 mins.

How I expect reviewers to test this PR:
- If the acceptance tests go 🟢  and the total time for running them is under 43 minutes for the partitions test, hit the like, comment, subscribe and merge button!!

### CODE REVIEW

File: acceptance/tests/partitions/partitions_test.go [hunk 1]
```diff
@@ -786,170 +359,280 @@ func TestPartitionsWithMesh(t *testing.T) {
 			helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
 				k8s.KubectlDeleteK(t, clientClusterContext.KubectlOptions(t), kustomizeDir)
 			})
+			// This section of the tests run the in-partition networking tests.
+			t.Cleanup(func() {
```
Comment (ishustava) [inline acceptance/tests/partitions/partitions_test.go:363]: why is it part of the cleanup?

File: acceptance/tests/partitions/partitions_test.go [hunk 2]
```diff
@@ -786,170 +359,280 @@ func TestPartitionsWithMesh(t *testing.T) {
...
+				// Client cluster.
+				services, _, err = consulClient.Catalog().Service(staticServerName, "", serverQueryClientOpts)
+				require.NoError(t, err)
+				require.Len(t, services, 1)
+
+				services, _, err = consulClient.Catalog().Service(staticClientName, "", clientQueryClientOpts)
+				require.NoError(t, err)
+				require.Len(t, services, 1)
+
+				defaultPartitionExports := &api.PartitionExportsConfigEntry{
```
Comment (ishustava) [inline acceptance/tests/partitions/partitions_test.go:533]: could we make it a CRD and put it in the fixture? I think this should be the workflow that we think users on k8s will use.

File: acceptance/tests/partitions/partitions_test.go (hunk 1 above)
Comment (thisisnotashwin) [reply acceptance/tests/partitions/partitions_test.go:363]: If I didnt not put it inside the cleanup block, the resources from this run would hang around and would could the x-partition tests to fail.

File: acceptance/tests/partitions/partitions_test.go (hunk 2 above)
Comment (thisisnotashwin) [reply acceptance/tests/partitions/partitions_test.go:533]: good call. yeah ill create a CRD fixture for this!

File: acceptance/tests/partitions/partitions_test.go (hunk 1 above)
Comment (ishustava) [reply acceptance/tests/partitions/partitions_test.go:363]: 🤔  Now it seems that most of our test logic lives in cleanup functions, so I'm wondering if we can reorder the test somehow to make it work?

Comment (thisisnotashwin) [reply acceptance/tests/partitions/partitions_test.go:363]: I thought the job of the cleanup functions was to invoke a cleanup when we reached the end 🤔 
I can think about this some more

**Review by t-eckert — APPROVED** [review]

File: acceptance/tests/partitions/partitions_test.go (hunk 1 above)
Comment (thisisnotashwin) [reply acceptance/tests/partitions/partitions_test.go]: swapping this for a `t.Run()` worked and I like it.

**Review by ishustava — APPROVED** [review]
🎉

**Pull request merged**